File: Python/matrix/basic_matrix_multiplication.py
class BasicMatrixMultiplication:
    @staticmethod
    def multiply(matrix_a, matrix_b):
        if len(matrix_a[0]) != len(matrix_b):
            raise ValueError("The dimensions of the matrices are not compatible for multiplication.")

        rows = len(matrix_a)
        cols = len(matrix_b[0])
        common = len(matrix_b)

        result = [[0.0 for _ in range(cols)] for _ in range(rows)]

        for i in range(rows):
            for j in range(cols):
                for k in range(common):
                    result[i][j] += matrix_a[i][k] * matrix_b[k][j]

        return result

File: Java/src/main/java/org/ulpgc/matrix/basic/BasicMatrixMultiplication.java
package org.ulpgc.matrix.basic;

public class BasicMatrixMultiplication {

    public double[][] multiply(double[][] matrixA, double[][] matrixB) {
        int rows = matrixA.length;
        int cols = matrixB[0].length;
        int common = matrixB.length;
        double[][] result = new double[rows][cols];

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                for (int k = 0; k < common; k++) {
                    result[i][j] += matrixA[i][k] * matrixB[k][j];
                }
            }
        }
        return result;
    }
}
